### DISCUSSION TITLE
Request deduplication

### DISCUSSION THREAD

**Discussion by enricoschaaf** [TanStack/query#553] [General] [answered]
Would it be possible to add request deduplication? Basically if a query with the same key is still pending, don' t send another request instead await the pending request and use it.

**Comment by tannerlinsley** [comment, marked as answer]
This should already be the case.

**Reply by enricoschaaf** [reply]
Nice, then the mistake is somewhere in my application. Thanks for this awesome library!

**Comment by tannerlinsley** [comment]
My pleasure!
